src/gpu/optixPrograms.cu

```diff
@@ -112,7 +115,7 @@ __forceinline__ __device__ void saveNonHitRayResult(float nonHitDistance)
 	displacement = {isnan(displacement.x()) ? 0 : displacement.x(), isnan(displacement.y()) ? 0 : displacement.y(),
 	                isnan(displacement.z()) ? 0 : displacement.z()};
 	Vec3f xyz = origin + displacement;
-	saveRayResult<false>(xyz, nonHitDistance, 0, RGL_ENTITY_INVALID_ID, Vec3f{NAN}, Vec3f{NAN}, 0.001f, Vec3f{NAN}, NAN);
+	saveRayResult<false>(xyz, nonHitDistance, 0, RGL_ENTITY_INVALID_ID, Vec3f{NAN}, Vec3f{NAN}, 0.0f, Vec3f{NAN}, NAN, 0.0f);
```

Comment:
Nice catch ;) Is there a reason that one before last parameter (incidentAngle) is set to NAN here, when other parameters of this type are set to 0.0f.

Reply:
Each field has been implemented individually and has its specifications to the non-hit value. We do not have a unification of non-hits by type. However, this is a good observation to consider in the future.